naming/src/main/java/com/alibaba/nacos/naming/utils/ServiceUtil.java

```diff
@@ -196,14 +196,14 @@ public static ServiceInfo selectInstancesWithHealthyProtection(ServiceInfo servi
             filteredResult.setHosts(allInstances);
             
             // will re-compute healthCount
-            long newHealthyCount = healthyCount;
-            if (originalTotal != allInstances.size()) {
-                for (com.alibaba.nacos.api.naming.pojo.Instance allInstance : allInstances) {
-                    if (allInstance.isHealthy()) {
-                        newHealthyCount++;
-                    }
+            long newHealthyCount = 0L;
+            // if (originalTotal != allInstances.size()) {
+            for (com.alibaba.nacos.api.naming.pojo.Instance allInstance : allInstances) {
+                if (allInstance.isHealthy()) {
```

Comment:
Please re-compute this count only when `allInstances` changed considering performance.

I think it just missing the `newHealthyCount = 0L;` in the `if` block started at line 200.

Reply:
sry,  have changed